pydantic_ai_slim/pydantic_ai/usage.py

```diff
@@ -71,11 +66,118 @@ def opentelemetry_attributes(self) -> dict[str, int]:
                     result[prefix + key] = value
         return result
 
+    def __repr__(self):
+        kv_pairs = (f'{f.name}={value!r}' for f in fields(self) if (value := getattr(self, f.name)))
+        return f'{self.__class__.__qualname__}({", ".join(kv_pairs)})'
+
     def has_values(self) -> bool:
         """Whether any values are set and non-zero."""
-        return bool(self.requests or self.request_tokens or self.response_tokens or self.details)
+        return any(dataclasses.asdict(self).values())
 
-    __repr__ = _utils.dataclasses_no_defaults_repr
+
+@dataclass(repr=False)
+class RequestUsage(UsageBase):
+    """LLM usage associated with a single request.
+
+    This is an implementation of `genai_prices.types.AbstractUsage` so it can be used to calculate the price of the
```

Comment:
Why it doesn't inherit the protocol?

Reply:
Pyright complains with things like `error: "input_tokens" incorrectly overrides property of same name in class "AbstractUsage"`

Reply:
If not even the one in `genai-prices` use it, what's the point? Is it even type safe if you use `AbstractUsage`?